avocado/utils/data_structures.py

```diff
@@ -243,3 +245,58 @@ def time_to_seconds(time):
     else:
         seconds = 0
     return seconds
+
+
+class DataSize(object):
+    """
+    Data Size object with builtin unit-converted attributes.
+
+    :param data: Data size plus optional unit string. i.e. '10Mb'. No
+                 unit string means the data size is in bytes.
+    """
+
+    def __init__(self, data):
+        self.__multipliers = OrderedDict([('b', 1),  # 2**0
+                                          ('k', 1024),  # 2**10
+                                          ('m', 1048576),  # 2**20
+                                          ('g', 1073741824),  # 2**30
+                                          ('t', 1099511627776)])  # 2**40
+
+        pattern = r"([0-9]+)([a-z]+)?"  # Number and optional string
+        self._value, unit = re.match(pattern,
+                                     data,
+                                     re.IGNORECASE).groups()
```

Comment:
Depending on the input type, this will return `None`, which won't have a `groups()` attribute, causing an attribute error exception.  It's better to raise an specific exception.

Reply:
Ok, let me improve the input type handling here.